### PR TITLE
[cleanup crusade] Shellcheck

### PR DESCRIPTION
**Pull request by paulfantom**
- Disabling shellcheck for deprecated shell modules
- Linting more shell modules
- lint installer, kickstart, docker/build.sh and .travis/containerized_build.sh

### CODE REVIEW

File: charts.d/libreswan.chart.sh [hunk 1]
```diff
@@ -143,7 +143,7 @@ libreswan_update_one() {
 
 	[ -z "${id}" ] && libreswan_create_one "${name}"
 
-	uptime=$(( libreswan_now - ${libreswan_established_add_time[${n}]} ))
+	uptime=$(( libreswan_now - libreswan_established_add_time[n] ))
```
Comment (ktsaou) [inline charts.d/libreswan.chart.sh]: I think `n` needs `${n}`:

```sh
declare -A x=([a]=1 [b]=2)
n="b"
echo $(( x[n] ))
0
echo $(( x[${n}] ))
2
```

File: charts.d/mysql.chart.sh [hunk 2]
```diff
@@ -56,7 +57,7 @@ mysql_check() {
 		shift
 	fi
 
-	[ -z "${mysql_cmd}" ] && mysql_cmd="$(which mysql 2>/dev/null || command -v mysql 2>/dev/null)"
+	[ -z "${mysql_cmd}" ] && mysql_cmd="$(command -v mysql 2>/dev/null)"
```
[the remark this answers is not in the document]
Comment (paulfantom) [reply charts.d/mysql.chart.sh]: I reverted that change.

File: charts.d/phpfpm.chart.sh [hunk 3]
```diff
@@ -192,4 +199,4 @@ EOF
 
 phpfpm_check
 phpfpm_create
-phpfpm_update
+phpfpm_update "$@"
```
Comment (ktsaou) [inline charts.d/phpfpm.chart.sh]: could you please remove these 3 lines?
they seem debug code.

**Comment by paulfantom** [comment]
Finished. All `*.sh` files are linted with shellcheck.

**Comment by paulfantom** [comment]
Closes #4166

File: charts.d/sensors.chart.sh [hunk 4]
```diff
@@ -139,66 +141,66 @@ sensors_create() {
 
 			case $mode in
 				temperature)
-					files="$( ls $path/temp*_input 2>/dev/null; ls $path/temp 2>/dev/null )"
-					files="$( sensors_check_files $files )"
-					files="$( sensors_check_temp_type $files )"
+					files="$( ls "$path/temp*_input" 2>/dev/null; ls "$path/temp" 2>/dev/null )"
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: `*` expansion does not happen in quoted strings:

```sh
ls -l "/etc/*"
ls: cannot access '/etc/*': No such file or directory
```

File: charts.d/sensors.chart.sh [hunk 5]
```diff
@@ -139,66 +141,66 @@ sensors_create() {
...
 					echo "CHART sensors.temp_$id '' '$name Temperature' 'Celsius' 'temperature' 'sensors.temp' line $((sensors_priority + 1)) $sensors_update_every"
-					echo >>$TMP_DIR/sensors.sh "echo \"BEGIN sensors.temp_$id \$1\""
+					echo >>"$TMP_DIR/sensors.sh" "echo \"BEGIN sensors.temp_$id \$1\""
 					divisor=1000
 					;;
 
 				voltage)
-					files="$( ls $path/in*_input 2>/dev/null )"
-					files="$( sensors_check_files $files )"
+					files="$( ls "$path/in*_input" 2>/dev/null )"
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: here too

File: charts.d/sensors.chart.sh [hunk 6]
```diff
@@ -96,34 +98,34 @@ sensors_check_temp_type() {
 
 # _create is called once, to create the charts
 sensors_create() {
-	local path= dir= name= x= file= lfile= labelname= labelid= device= subsystem= id= type= mode= files= multiplier= divisor=
+	local path='' dir='' name='' x='' file='' lfile='' labelname='' device='' subsystem='' id='' type='' mode='' files='' multiplier='' divisor=''
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: I thought variables assignment on `local` variables definition is wrong.

File: charts.d/sensors.chart.sh [hunk 7]
```diff
@@ -139,66 +141,66 @@ sensors_create() {
...
 					echo "CHART sensors.volt_$id '' '$name Voltage' 'Volts' 'voltage' 'sensors.volt' line $((sensors_priority + 2)) $sensors_update_every"
-					echo >>$TMP_DIR/sensors.sh "echo \"BEGIN sensors.volt_$id \$1\""
+					echo >>"$TMP_DIR/sensors.sh" "echo \"BEGIN sensors.volt_$id \$1\""
 					divisor=1000
 					;;
 
 				current)
-					files="$( ls $path/curr*_input 2>/dev/null )"
-					files="$( sensors_check_files $files )"
+					files="$( ls "$path/curr*_input" 2>/dev/null )"
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: and here

File: charts.d/sensors.chart.sh [hunk 8]
```diff
@@ -139,66 +141,66 @@ sensors_create() {
...
 					echo "CHART sensors.curr_$id '' '$name Current' 'Ampere' 'current' 'sensors.curr' line $((sensors_priority + 3)) $sensors_update_every"
-					echo >>$TMP_DIR/sensors.sh "echo \"BEGIN sensors.curr_$id \$1\""
+					echo >>"$TMP_DIR/sensors.sh" "echo \"BEGIN sensors.curr_$id \$1\""
 					divisor=1000
 					;;
 
 				power)
-					files="$( ls $path/power*_input 2>/dev/null )"
-					files="$( sensors_check_files $files )"
+					files="$( ls "$path/power*_input" 2>/dev/null )"
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: here too

File: charts.d/sensors.chart.sh [hunk 9]
```diff
@@ -139,66 +141,66 @@ sensors_create() {
...
 					echo "CHART sensors.power_$id '' '$name Power' 'Watt' 'power' 'sensors.power' line $((sensors_priority + 4)) $sensors_update_every"
-					echo >>$TMP_DIR/sensors.sh "echo \"BEGIN sensors.power_$id \$1\""
+					echo >>"$TMP_DIR/sensors.sh" "echo \"BEGIN sensors.power_$id \$1\""
 					divisor=1000000
 					;;
 
 				fans)
-					files="$( ls $path/fan*_input 2>/dev/null )"
-					files="$( sensors_check_files $files )"
+					files="$( ls "$path/fan*_input" 2>/dev/null )"
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: here too

File: charts.d/sensors.chart.sh [hunk 10]
```diff
@@ -139,66 +141,66 @@ sensors_create() {
...
 					[ -z "$files" ] && continue
 					echo "CHART sensors.fan_$id '' '$name Fans Speed' 'Rotations / Minute' 'fans' 'sensors.fans' line $((sensors_priority + 5)) $sensors_update_every"
-					echo >>$TMP_DIR/sensors.sh "echo \"BEGIN sensors.fan_$id \$1\""
+					echo >>"$TMP_DIR/sensors.sh" "echo \"BEGIN sensors.fan_$id \$1\""
 					;;
 
 				energy)
-					files="$( ls $path/energy*_input 2>/dev/null )"
-					files="$( sensors_check_files $files )"
+					files="$( ls "$path/energy*_input" 2>/dev/null )"
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: and here

File: charts.d/sensors.chart.sh [hunk 11]
```diff
@@ -139,66 +141,66 @@ sensors_create() {
...
-					echo >>$TMP_DIR/sensors.sh "echo \"BEGIN sensors.energy_$id \$1\""
+					echo >>"$TMP_DIR/sensors.sh" "echo \"BEGIN sensors.energy_$id \$1\""
 					algorithm="incremental"
 					divisor=1000000
 					;;
 
 				humidity)
-					files="$( ls $path/humidity*_input 2>/dev/null )"
-					files="$( sensors_check_files $files )"
+					files="$( ls "$path/humidity*_input" 2>/dev/null )"
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: and here

File: charts.d/sensors.chart.sh [hunk 12]
```diff
@@ -211,26 +213,27 @@ sensors_create() {
...
 				file="$x"
 				fid="$( fixid "$file" )"
-				lfile="$( basename $file | sed "s|_input$|_label|g" )"
-				labelname="$( basename $file | sed "s|_input$||g" )"
+				lfile="$( basename "$file" | sed "s|_input$|_label|g" )"
+				labelname="$( basename "$file" | sed "s|_input$||g" )"
 
-				if [ ! "$path/$lfile" = "$file" -a -f "$path/$lfile" ]
+				if [ ! "$path/$lfile" = "$file" ] && [ -f "$path/$lfile" ]
 					then
```
Comment (ktsaou) [inline charts.d/sensors.chart.sh]: indentation

File: charts.d/squid.chart.sh [hunk 13]
```diff
@@ -51,14 +50,16 @@ squid_check() {
 	require_cmd sed || return 1
 	require_cmd egrep || return 1
 
-	if [ -z "$squid_host" -o -z "$squid_port" -o -z "$squid_url" ]
+	if [ -z "$squid_host" ] || [ -z "$squid_port" ] || [ -z "$squid_url" ]
 		then
```
Comment (ktsaou) [inline charts.d/squid.chart.sh]: indentation

File: charts.d/squid.chart.sh [hunk 14]
```diff
@@ -113,8 +114,8 @@ squid_update() {
 	# even if something goes wrong, no other code can be executed
 
 	eval "$(squid_get_stats |\
-		 sed -e "s/ \+/ /g" -e "s/\./_/g" -e "s/^\([a-z0-9_]\+\) *= *\([0-9]\+\)$/local squid_\1=\2/g" |\
-		egrep "^local squid_(client_http|server_all)_[a-z0-9_]+=[0-9]+$")"
+		 sed -e "s/ \\+/ /g" -e "s/\\./_/g" -e "s/^\\([a-z0-9_]\\+\\) *= *\\([0-9]\\+\\)$/local squid_\\1=\\2/g" |\
```
Comment (ktsaou) [inline charts.d/squid.chart.sh]: why?

Comment (paulfantom) [reply charts.d/squid.chart.sh]: https://github.com/koalaman/shellcheck/wiki/SC1117

Comment (paulfantom) [reply charts.d/squid.chart.sh]: Due to https://github.com/koalaman/shellcheck/wiki/SC1117 which should be retired, but isn't in latest shellcheck version

Comment (ktsaou) [reply charts.d/squid.chart.sh]: I don't think it will work. Have your tried it?

Comment (paulfantom) [reply charts.d/squid.chart.sh]: No. I will revert that.

**Pull request merged**